ts/src/base/Exchange.ts

```diff
@@ -841,7 +841,7 @@ export default class Exchange {
                     httpProxyAgent = this.httpAgent;
                 }
             }
-            url = proxyUrl + url;
+            url = proxyUrl + this.encodeURIComponent (url); 
```

Comment:
@ttodua can we put in this a function like formatProxyUrl, so the users can monkey patch it if needed?

Reply:
for monkeypatching, users already have `proxyUrlCallback` where they have full control, so extra method is not needed here

Reply:
ah, i was mistaken, `proxyUrlCallback` only affects proxy itself. so, ok, i can do that, jfyi, that user also said that only urlencoding was needed and it would fix the issue: https://github.com/ccxt/ccxt/issues/25574#issuecomment-2775435385

so maybe there is no real-world scenario where users might need to monkeypatch this, wdyt? as you say

Reply:
@ttodua another thing, you're URIEncoding the entire URL and not the query right? not sure if that's safe

Reply:
anyway, I've added now a full callback and users can override it